src/service-catalog.ts

```diff
@@ -63,107 +57,137 @@ export interface DescribeProvisionedProductOptions {
-  } while (token);
+function resolveProvisioningArtifact(provisioningArtifacts: AWS.ServiceCatalog.ProvisioningArtifacts, provisioningArtifactId?: string):
+AWS.ServiceCatalog.ProvisioningArtifact {
+  if (provisioningArtifactId) {
+    return provisioningArtifacts.filter(pa => pa.Id == provisioningArtifactId).pop()!;
+  } else if (provisioningArtifacts.filter(pa => pa.Guidance == 'DEFAULT').length == 1) {
+    return provisioningArtifacts.filter(pa => pa.Guidance == 'DEFAULT').pop()!;
+  } else {
+    return provisioningArtifacts.sort((a, b) => a.CreatedTime!.valueOf() - b.CreatedTime!.valueOf()).pop()!;
+  }
+}
 
-  return products;
+/**
+ * Returns the queried launch path from list of available launch paths.
+ * @param launchPaths list of available launch paths for the product
+ * @param launchPathId the query launch path's Id
+ * @returns the launch path summary for the query launch path
+ */
+function resolveLaunchPath(launchPaths: AWS.ServiceCatalog.LaunchPathSummaries, launchPathId?: string): AWS.ServiceCatalog.LaunchPathSummary {
+  if (launchPathId) {
+    return launchPaths.filter(lp => lp.Id == launchPathId).pop()!;
+  } else if (launchPaths.length == 1) {
+    return launchPaths.pop()!;
+  } else {
+    throw new Error('Unable to resolve between multiple launch paths.');
+  }
 }
 
 /**
- * Fetches the all the available or queried provisioning artifact(s) for the caller.
+ * Fetches the all the available or queried service catalog product(s) for the caller.
  *
- * @param productId the product to fetch provisioning artifacts for
- * @returns list of provisioning artifact details for a product
+ * @returns list of product view summaries
  */
-export async function listProvisioningArtifacts(
-  productId: string,
-  options: DescribeProvisionedProductOptions = {}): Promise<AWS.ServiceCatalog.ProvisioningArtifactDetail[]> {
-  const sc = options.client ?? new ServiceCatalogClient();
+async function describeProduct(options: DescribeProductAggregateOptions): Promise<ProductDataAggregate> {
+  const sc: IServiceCatalogClient = options.client ?? new ServiceCatalogClient();
 
-  const provisioningArtifacts: AWS.ServiceCatalog.ListProvisioningArtifactsOutput = await sc.listProvisioningArtifacts({
-    ProductId: productId,
+  const describeProductResponse: AWS.ServiceCatalog.DescribeProductOutput = await sc.describeProduct({
+    Id: options.productId,
   });
 
-  if ( options.provisioningArtifactId != null ) {
-    return provisioningArtifacts.ProvisioningArtifactDetails!.filter(pa => pa.Id == options.provisioningArtifactId);
-  } else {
-    return provisioningArtifacts.ProvisioningArtifactDetails!;
-  }
+  const provisioningArtifact: AWS.ServiceCatalog.ProvisioningArtifact = resolveProvisioningArtifact(describeProductResponse.ProvisioningArtifacts!,
+    options.provisioningArtifactId);
+
+  const launchPath: AWS.ServiceCatalog.LaunchPathSummary = resolveLaunchPath(describeProductResponse.LaunchPaths!, options.launchPathId);
```

Comment:
Will `LaunchPaths` always be defined?

Reply:
LaunchPaths will always be defined.

A LaunchPath signifies the Portfolio (and constraints) an end user is provisioning through.

The only way an end user has access to a product is through a portfolio, so if you have a product available it has to have a launch path

Reply:
handled.